internal/engine/experiment/tor/tor.go

```diff
@@ -293,15 +287,16 @@ func maybeSanitize(input TargetResults, kt keytarget) TargetResults {
 func (rc *resultsCollector) measureSingleTarget(
 	ctx context.Context, kt keytarget, total int,
 ) {
-	tk, err := rc.flexibleConnect(ctx, kt)
+	tk, failure := rc.flexibleConnect(ctx, kt)
+	runtimex.PanicIfNil(tk, "measurex should guarantee non-nil here")
```

Comment:
Here is where I call `flexibleConnect`. Because the "easy" API provides a strong guarantee that it does not return a `nil` structure, I've added an assertion to catch the error. (Having worked a bit with type-aware languages like Rust and Kotlin recently, the treatment of `nil` in Go makes me a bit sad 🤷.)